src/Models/tests/DetailsModelTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+declare(strict_types=1);
+//namespace Transformers\Models\tests;
+require_once '../DetailsModel.php';
+use Transformers\Models\DetailsModel;
+require_once '../../Abstracts/Transformer.php';
+use Transformers\Abstracts\Transformer;
+use PHPUnit\Framework\TestCase;
+
+
+class DetailsModelTest extends TestCase
+{
+    public function testSetTransformer_ThrowsException_GivenInvalidTypeInt()
+    {
+        $DetailsModel = new DetailsModel();
+        $this->expectException(TypeError::class);
+        $result = $DetailsModel->setTransformer(22);
+    }
+    public function testSetTransformer_ThrowsException_GivenInvalidTypeStr()
+    {
+        $DetailsModel = new DetailsModel();
+        $this->expectException(TypeError::class);
+        $result = $DetailsModel->setTransformer('badstr');
+    }
+    public function testSetTransformer_ThrowsException_GivenInvalidTypeArr()
+    {
+        $DetailsModel = new DetailsModel();
+        $this->expectException(TypeError::class);
+        $result = $DetailsModel->setTransformer(['bad', 'array']);
+    }
+    public function testSetTransformer_ThrowsException_GivenInvalidTypeBool()
+    {
+        $DetailsModel = new DetailsModel();
+        $this->expectException(TypeError::class);
+        $result = $DetailsModel->setTransformer(true);
+    }
+    public function testSetTransformer_ThrowsException_GivenInvalidTypeNull()
+    {
+        $DetailsModel = new DetailsModel();
+        $this->expectException(TypeError::class);
+        $result = $DetailsModel->setTransformer(null);
+    }
+    public function testSetTransformer_ThrowsException_GivenInvalidObject()
+    {
+        $DetailsModel = new DetailsModel();
+        $stub = $this->getMockBuilder(\stdclass::class)->getMock();
+        $this->expectException(TypeError::class);
+        $result = $DetailsModel->setTransformer($stub);
+    }
+    public function testSetTransformer_ThrowsException_GivenValidObject()
```

Comment:
This test needs renaming as there is no exception thrown.

Reply:
Renamed.